### ISSUE TITLE
Investigate using pip-tools to help us with python deps

### ISSUE THREAD

**Issue by nicksellen** [yunity/yunity-core#126]
Problem: we have exact versions pinned in requirements.pip so when we want to update the deps, there is no easy way

It is a good idea to have the exact versions pinned, but it does make this part trickier.

pip-tools to the rescue ---> https://koed00.github.io/managed-environments-with-piptools/

... might be useful

**Comment by tiltec** [comment]
The pip-sync command looks really helpful.
I suppose pip-tools need be installed outside of the virtualenv, otherwise we would have an chicken-and-egg problem :)

**Comment by benjaoming** [comment]
Hi, just passing by to see your project :) You could chip in to the testing of this new pip command: https://github.com/pypa/pip/pull/2492

Good luck with the hackathon!

**Comment by NerdyProjects** [comment]
I regenerated the requirements using pip-compile and documented that in 4139464171d58aa3d7b1383bad50dec106bf2892
I did not yet switch to pip-sync although this seems good for a production environment? Would then also justify to include pip-tools in the requirements. Then, a pip install followed by a pip-sync seems reasonable :)
@benjaoming: I tried that - Seems to work, but I don't see a use case for us.
